heartbeat/monitors/active/http/respbody.go

```diff
@@ -28,30 +28,26 @@ import (
 
 	"github.com/elastic/beats/v7/heartbeat/reason"
 	"github.com/elastic/beats/v7/libbeat/common"
+	"github.com/elastic/beats/v7/libbeat/mime"
 )
 
 // maxBufferBodyBytes sets a hard limit on how much we're willing to buffer for any reason internally.
 // since we must buffer the whole body for body validators this is effectively a cap on that.
 // 100MiB out to be enough for everybody.
 const maxBufferBodyBytes = 100 * units.MiB
 
-func processBody(resp *http.Response, config responseConfig, validator multiValidator) (common.MapStr, reason.Reason) {
+func processBody(resp *http.Response, config responseConfig, validator multiValidator) (common.MapStr, string, reason.Reason) {
 	// Determine how much of the body to actually buffer in memory
-	var bufferBodyBytes int
-	if validator.wantsBody() {
-		bufferBodyBytes = maxBufferBodyBytes
-	} else if config.IncludeBody == "always" || config.IncludeBody == "on_error" {
+	bufferBodyBytes := maxBufferBodyBytes
+	if !validator.wantsBody() && (config.IncludeBody == "always" || config.IncludeBody == "on_error") {
```

Comment:
So, this introduces a subtle behavior change in that it breaks the body validators, only letting them validate against the specified (or default) portion of the body. Those validators do things like look for the presence of a string to match (against up to the first 100MiB).

We should restore the initial `if validator.wantsBody()` clause. The `else` clause should indeed have been removed.

Reply:
👍 reverted the conditional and set a minimum buffer threshold so we can analyze the body still.